include/vast/Dialect/HighLevel/HighLevelOps.td

```diff
@@ -29,7 +26,7 @@ def TranslationUnitOp
   let assemblyFormat = [{ $body attr-dict }];
 }
 
-def ScopeOp : HighLevel_Op< "scope", [NoTerminator] >
+def ScopeOp : HighLevel_Op< "scope", [NoTerminator, DeclareOpInterfaceMethods<RegionKindInterface>] >
```

Comment:
I know this is not part of this PR, but ScopeOp should be part of Core, as it is not actual AST primitive.

Reply:
What do you mean by this? That in codegen we should emit `core.scope` instead of `hl.scope`?

Reply:
Okay, I will do it in a separate PR as this is already getting quite big?